### ISSUE TITLE
Add empty summary tags to all objects, properties and methods.

### ISSUE THREAD

**Issue by tacosontitan** [ObsidianMC/Obsidian#135]
After reviewing a large chunk of files, I've noticed that summary tags are not present to enable natural understanding and reduce the likelihood of open interpretation when working with objects, properties or methods. I would like to propose that summary tags be added to:

 - User-Defined Types
 - Properties
 - Methods

Additionally, if exceptions can be thrown from within methods or the `get`/`set` methods of properties then the appropriate exception tags should also be added.

---
At a minimum, these tags should be present, even if empty to consider this issue completed. This is because having the tags present will *help* encourage developers that have an understanding of the area they're working in, to add documentation for intellisense to pick up on. It also enables the ability to generate living documentation for the API project.

**Comment by Craftplacer** [comment]
I would suggest, don't over-document the code base. There are cases (applicable in any C# project) where the method name and class context are more valuable and self-explanatory than a `<summary>Gets a player</summary>`.

I don't agree on the "add empty summaries everywhere" as they're as useless as not adding a summary at all, in my opinion.

I would rephrase this issue as "Add missing code documentation" (XMLdocs).

**Comment by Craftplacer** [comment]
Also at further investigation, this issue seems a duplicate of #79.

@Naamloos any interest of merging/extending the scope of #79 to the entire codebase?

**Comment by Naamloos** [comment]
Suresies, big yes.

Documentation and comments are important, and obsidian is severely lacking in it.
I'd say, add summaries to everything where it'd make sense.

**Comment by Craftplacer** [comment]
Should we merge this issue into #79 and close it then?

**Comment by tacosontitan** [comment]
> Should we merge this issue into #79 and close it then?

Ah, didn't see that in my searches; I'd concur with the merge and closure.

**Comment by tacosontitan** [comment]
Oh, I forgot that I can close it :)
